src/pages/Profile/index.js

```diff
@@ -1,36 +1,75 @@
 import cx from 'classnames';
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import ReactGA from 'react-ga'
 import Avatar from '../../components/Avatar';
 import TimeAgo from 'react-timeago';
+import TotalWords from '../../components/TotalWords';
+import { userPreferencesUpdateRequestAction } from '../../actions/userPreferences';
 
-export default class Profile extends Component {
-
+class Profile extends Component {
   constructor(props) {
     super(props);
 
     const { user } = props || {};
+    const { preferences } = props;
     this.state = {
       success: '',
-      displayName: user.displayName
+      displayName: user.displayName,
+      totalWords: preferences.totalWords || 5,
     };
   }
 
   componentDidMount() {
     ReactGA.pageview('/profile');
   }
 
+  componentDidUpdate(prevProps) {
+    const { preferences } = this.props;
+
+    if (prevProps.preferences.totalWords !== preferences.totalWords) {
+      this.setState({ totalWords: preferences.totalWords });
+    }
+  }
+
   async doUpdateProfile(event) {
     event.preventDefault();
     const { user } = this.props;
-    await user.updateProfile({
-      displayName: this.state.displayName
-    });
+    const { displayName } = this.state;
+    await user.updateProfile({ displayName });
     this.setState({
       success: 'Profile updated'
     });
   }
 
+  async doUpdatePreferences(event) {
+    event.preventDefault();
+    const { user, updateUserPreferences } = this.props;
+    const { totalWords } = this.state;
+    updateUserPreferences(user.uid, { totalWords });
+    this.setState({
+      success: 'Preferences updated',
+      totalWords,
+    });
+  }
+
+  increment = () => {
+    const { totalWords } = this.state;
+    this.setState({
+      totalWords: totalWords + 1,
+    });
+  }
+
+  decrement = () => {
+    const { totalWords } = this.state;
+
+    if (totalWords > 1) {
+      this.setState({
+        totalWords: totalWords - 1,
+      });
+    }
+  }
```

Comment:
Can we bind `increment` and `decrement` in the constructor instead of using the class-property notation? There's even an handy `utils/bind` function you can use

Reply:
@clayreimann I made this change.  This article https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56 points that the use of the property notation to have most advantages. Perhaps we should decide to use that notation throughout?